hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala

```diff
@@ -217,9 +217,6 @@ class ServiceBackend(
     stageCount += 1
     implicit val formats: Formats = DefaultFormats
     val batchState = (batch \ "state").extract[String]
-    if (batchState == "failed") {
-      throw new HailBatchFailure(s"Update $updateId for batch $batchId failed")
-    }
```

Comment:
@daniel-goldstein you have a bit more recent context on the batch API than I do; can you help Tim get this to the right place?

I think what we want here is to still check for batchState failure, but instead of throwing an exception unconditionally, we want to ask batch which job failed and then try to read that job's result file. If the result file doesn't exist, we probably had the JVM crash.

Reply:
My mental model of my change in this file is that instead of throwing an exception unconditionally, I throw an exception when we get to the first failing job (HailUserException or otherwise). Is that not what's going on?

How is this changing what exceptions propagate to Python? Is something else looking for failing statuses in Python and ignoring the propagated user error if the batch fails?

Reply:
My guess for why the tests are failing here is because this difference in exception handling:

https://github.com/hail-is/hail/blob/35e7c4915377da3264955215da2b389a15e18fa6/hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala#L691-L701

Reply:
Er, actually, not entirely sure.